scripts/build/index.ts

```diff
@@ -6,6 +6,7 @@ import fs from 'node:fs/promises';
 import esMain from 'es-main';
 import stringify from 'fast-json-stable-stringify';
 import { compareVersions } from 'compare-versions';
+import { marked } from 'marked';
```

Comment:
I'd strongly discourage you from doing this specific thing. Marked doesn't fully implement CommonMark. If you ever try to do something else with Markdown or consume the Markdown using another tool (e.g., an off the shelf linter), you'll be very sad. Consider using something like [remark and remark-rehype](https://remark.js.org/) instead.

Reply:
Do you happen to know what parts of CommonMark this library doesn't implement?

Reply:
> Do you happen to know how its output differs from the spec?

Not specifically. The [marked docs](https://marked.js.org/#specifications) link to a discussion that was last updated in 2022, but several areas (including links) don't pass the CommonMark tests, but it doesn't show specifically which tests were failing.

> it may not be something to worry about

This is how it _always_ starts. I've been on no less than three projects where somebody said, "we're not using this for anything big, it'll be no problem to use a non-standard Markdown implementation" and before long someone finds another use for the non-standard Markdown parser and then, finally, the day arrives where it becomes a problem that requires substantial rework to fix. Please don't let me be the Cassandra of Markdown.

It's so much easier to do the right thing when it's not a big deal than it is to retrofit an existing body of work.

Reply:
I cloned the repository locally and made some modifications to determine what exactly is failing in the tests for `marked`.  I have [uploaded a file containing the (simplified) output log of the failing tests](https://github.com/user-attachments/files/16344340/marked-failing-commonmark-tests.txt).

To summarize the results, here's what I've found:

Examples 32, 33 and 503 are about encoding characters with accents a specific way within URLs (`f&ouml;&ouml;` (`föö`) becomes `f%C3%B6%C3%B6`).  It appears that `marked` decodes characters with accents, causing these examples to fail tests.  However, since browsers now automatically encode URLs, I believe this is a non-issue.

Example 507 has to do with the specific white spacing needed to parse links with titles.  The CommonMark spec requires only spaces, tabs or single line endings between the link and any title text, otherwise it should all be included as a part of the URL.  `marked` treats a non-breaking space as valid whitespace.

Examples 497, 512, 518-520, 524, 526, 528, 532-533, 536 and 538 are basically how the parser should handle syntax errors related to links (such as overlapping syntax usage, like `[foo [bar](/uri)](/uri)`).

Example 540 has to do with encoding/decoding non-ASCII characters in reference links (`ß` <-> `SS`).  For BCD's use case, this is a non-issue, as we do not use reference links anywhere in BCD (and would not be able to in notes/descriptions anyways).

Examples 573-577, 585 and 589 are all about stripping out Markdown syntax from the alt text in images.  For BCD, this is again a non-issue, as images are not permitted in notes/descriptions.

Reply:
For your consideration https://github.com/markdown-it/markdown-it  (via [Eleventy](https://www.11ty.dev/docs/languages/markdown/))

Reply:
Let's be pragmatic and choose `marked`. We can reconsider when we actually have an issue, but afaik for now this will be used almost exclusively for links, bold, and italic.